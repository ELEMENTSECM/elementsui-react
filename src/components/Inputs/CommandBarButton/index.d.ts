import * as React from 'react';

export interface CommandBarButtonProps {
	/**
	 * Button label
	 */
	text?: string;
	/**
	 * Button is disabled
	 */
	disabled?: boolean;
	/**
	 * Mouse click event handler
	 */
	onClick?: (...args: any[]) => any;
	/**
	 * Icon name
	 */
	iconName?: string;
	/**
	 * User-defined styling
	 */
	styles?: (...args: any[]) => any;
}

declare const CommandBarButton: React.SFC<CommandBarButtonProps>;

export default CommandBarButton;
