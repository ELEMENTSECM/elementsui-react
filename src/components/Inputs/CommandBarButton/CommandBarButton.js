import * as React from 'react';
import PropTypes from 'prop-types';
import { CommandBarButton as FabricCommandBarButton } from 'office-ui-fabric-react/lib/Button';
import { styles } from './CommandBarButton.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

/** CommandBarButton example */
export function CommandBarButton(props) {
	const { text, disabled, onClick, iconName, styles } = props;
	const classNames = classNamesFunction()(styles, props);
	return (
		<FabricCommandBarButton
			className={classNames.root}
			text={text}
			disabled={disabled}
			onClick={onClick}
			iconProps={{ iconName: iconName }}
		/>
	);
}

CommandBarButton.propTypes = {
	/** Button label */
	text: PropTypes.string,
	/** Button is disabled */
	disabled: PropTypes.bool,
	/** Mouse click event handler */
	onClick: PropTypes.func,
	/** Icon name */
	iconName: PropTypes.string,
	/** User-defined styling */
	styles: PropTypes.func
};

export default styled(customizable('CommandBarButton', ['theme'])(CommandBarButton), styles);
