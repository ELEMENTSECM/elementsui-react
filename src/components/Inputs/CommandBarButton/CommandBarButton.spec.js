import React from 'react';
import renderer from 'react-test-renderer';
import CommandBarButton from './CommandBarButton';

describe('CommandBarButton', () => {
	test('displays `Foo` as text', () => {
		const tree = renderer.create(<CommandBarButton text="Foo" icon="Color" />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
