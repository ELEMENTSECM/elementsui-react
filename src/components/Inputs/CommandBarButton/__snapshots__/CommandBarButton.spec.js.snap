// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandBarButton displays \`Foo\` as text 1`] = `
<button
  aria-describedby={null}
  aria-label={undefined}
  aria-labelledby={null}
  aria-pressed={undefined}
  className="ms-Button ms-Button--commandBar root-38"
  data-is-focusable={true}
  disabled={undefined}
  onClick={undefined}
  type="button"
>
  <div
    className="ms-Button-flexContainer flexContainer-39"
  >
    <i
      aria-hidden={true}
      className="root-46 ms-Button-icon icon-41"
      data-icon-name={undefined}
      role="presentation"
    />
    <div
      className="ms-Button-textContainer textContainer-40"
    >
      <div
        className="ms-Button-label label-42"
        id="id__0"
      >
        Foo
      </div>
    </div>
  </div>
</button>
`;
